src/twisted/conch/client/default.py

```diff
@@ -130,6 +130,31 @@ def isInKnownHosts(host, pubKey, options):
 
 
 
+def getHostKeyAlgorithms(host, options):
```

Comment:
Maybe this could have been left as an internal API.  Maybe there could be other clients outside of the conch client that could guess at a preferred host key type based on what is in known_hosts